content/en/docs/examples/virtual-machines/autoregistration/index.md

```diff
@@ -0,0 +1,163 @@
+---
+title: Automated WorkloadEntry creation with WorkloadGroup
+description: Learn how to use the experimental automated VM registration feature.
+  Istio mesh.
+weight: 20
+keywords:
+- kubernetes
+- vms
+- virtual-machines
+owner: istio/wg-environments-maintainers
+test: no
+---
+
+{{< warning >}}
+This feature is experimental in Istio 1.8
+{{< /warning >}}
+
+This example provides instructions to integrate a virtual machine, or a bare metal host into a
+single network Istio mesh deployed on Kubernetes without manual WorkloadEntry creation. This approach requires L3 connectivity
+between the virtual machine, and the Kubernetes cluster.
+
+## Prerequisites
+
+- One or more Kubernetes clusters with versions: {{< supported_kubernetes_versions >}}.
+
+- Virtual machines must have L3 IP connectivity to the endpoints in the mesh.
+  This typically requires a VPC or a VPN, as well as a container network that
+  provides direct (without NAT or firewall deny) routing to the endpoints. The
+  machine is not required to have access to the cluster IP addresses assigned by
+  Kubernetes.
+
+- Installation must be completed using [virtual machine installation](/docs/setup/install/virtual-machine) instructions.
+
+## Prepare the guide environment
+
+Set the environment variables `VM_NAMESPACE` and `SERVICE_ACCOUNT` (use the same values as during the installation guide):
+
+{{< text bash >}}
+$ VM_NAMESPACE="<the name of your service namespace>"
+$ SERVICE_ACCOUNT="<name of the Kubernetes service account you want to use for your VM>"
+{{< /text >}}
+
+### Running services on the virtual machine
+
+1. Setup an HTTP server on the virtual machine to serve HTTP traffic on port 8080:
+
+    {{< text bash >}}
+    $ python -m SimpleHTTPServer 8080
+    {{< /text >}}
+
+    {{< warning >}}
+    You may have to open firewalls to be able to access the 8080 port on your virtual machine
+    {{< /warning >}}
+
+1. Add an associated Service to the mesh
+
+    {{< text bash >}}
+    $ cat <<EOF | kubectl -n "${VM_NAMESPACE}" apply -f -
+    apiVersion: v1
+    kind: Service
+    metadata:
+      name: auto-cloud-vm
+      labels:
+        app: auto-cloud-vm
+    spec:
+      ports:
+      - port: 8080
+        name: http-vm
+        targetPort: 8080
+      selector:
+        app: auto-cloud-vm
+    EOF
+    {{< /text >}}
+
+## Configure VM for Auto-Registration
+
+1. Create the auto-registration group.
```

Comment:
the mainline docs should probably just be modified to use this. we can incrementally change it though I think